### PR TITLE
 Storage Write API Client library and configs addition

### PR DESCRIPTION
**Pull request by b-goyal**
This PR updates Pom file to include client library and required configs for the new API. 
Includes - 
1. google cloud bom as documented here : https://cloud.google.com/bigquery/docs/reference/storage/libraries#install 
2. Cleanup of pinned protobuf-java and auth dependency as the latest bigquery version brings in  these.
3. Configs required from one-pager https://confluentinc.atlassian.net/wiki/spaces/CONNECT/pages/2961999467/Storage+Write+API+in+BQ+Sink+Connector#StorageWriteAPIinBQSinkConnector-User-FacingChanges 

JIRA tickets : https://confluentinc.atlassian.net/browse/CC-19645 , https://confluentinc.atlassian.net/browse/CC-19657 

Epic : https://confluentinc.atlassian.net/browse/CC-19644

### CODE REVIEW

**Comment by pbadani** [comment]
is this a new minor version?

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkConfig.java [hunk 1]
```diff
@@ -329,6 +329,29 @@ public class BigQuerySinkConfig extends AbstractConfig {
       + "tables, and periodic merge flushes. Row-matching will be performed based on the contents " 
       + "of record keys.";
 
+  public static final String USE_STORAGE_WRITE_API_CONFIG = "useStorageWriteApi";
+
+  private static final ConfigDef.Type USE_STORAGE_WRITE_API_TYPE = ConfigDef.Type.BOOLEAN;
+  public static final boolean USE_STORAGE_WRITE_API_DEFAULT = true;
```
Comment (pbadani) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkConfig.java:335]: the default behaviour changes with this one

**Review by pbadani — APPROVED** [review]
LGTM with a minor comment.

**Comment by b-goyal** [comment]
> is this a new minor version?

yes

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkConfig.java (hunk 1 above)
Comment (b-goyal) [reply kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkConfig.java:335]: Thought of keeping this true till we are release ready so all tests run with this config true.

File: pom.xml [hunk 2]
```diff
@@ -132,6 +128,13 @@
 
     <dependencyManagement>
         <dependencies>
+            <dependency>
+                <groupId>com.google.cloud</groupId>
+                <artifactId>libraries-bom</artifactId>
+                <version>26.11.0</version>
```
Comment (aniketshrimal) [inline pom.xml:134]: Instead of adding version explicitly here - lets create a variable in properties section and use that here.

**Review by aniketshrimal — APPROVED** [review]
LGTM @b-goyal  - just one minor nit comment. Thanks!

**Pull request merged**